utility_network/display-utility-associations/src/main/java/com/esri/samples/display_utility_associations/DisplayUtilityAssociationsSample.java

```diff
@@ -0,0 +1,245 @@
+          }
+          // Add all junctions to the map.
+          else if (nSource.getSourceType() == UtilityNetworkSource.Type.JUNCTION) {
+            mapView.getMap().getOperationalLayers().add(new FeatureLayer(nSource.getFeatureTable()));
+          }
+          
+        // add association graphics at the initial view point
+        addAssociation();
+        // listen for navigation changes
+        mapView.addNavigationChangedListener((event) -> {
+          addAssociation();
+        });
+      });
+
+      // create ImageView for teo symbol
+      ImageView attachmentImageView = new ImageView();
+      ImageView connectivityImageView = new ImageView();
+      
+      // Create a graphics overlay for associations.
+      associationsOverlay = new GraphicsOverlay();
+      
+      // Create Labels for the preview.
+      Label attachmentLabel = new Label("Attachment");
+      Label connectivityLabel = new Label("Connectivity");
+  
+      // Create Symbols for the associations and preview.
+      SimpleLineSymbol attachmentSymbol = new SimpleLineSymbol(SimpleLineSymbol.Style.DOT, 0xFF00FF00, 5);
+      SimpleLineSymbol connectivitySymbol = new SimpleLineSymbol(SimpleLineSymbol.Style.DOT, 0xFFFF0000, 5);
+  
+      // get the symbol from the attachmentSymbol
+      ListenableFuture<Image> attachmentBitmap = attachmentSymbol.createSwatchAsync(0x00000000);
+      attachmentBitmap.addDoneListener(() -> {
+        try {
+          // display the image view in the preview area
+          attachmentImageView.setImage(attachmentBitmap.get());
+        } catch (InterruptedException | ExecutionException e) {
+          new Alert(Alert.AlertType.ERROR, "Error creating preview ImageView from provided attachmentSymbol" + e.getMessage()).show();
+        }
+      });
+  
+      // get the symbol from the connectivitySymbol
+      ListenableFuture<Image> connectivityBitmap = connectivitySymbol.createSwatchAsync(0x00000000);
+      connectivityBitmap.addDoneListener(() -> {
+        try {
+          // display the image view in the preview area
+          connectivityImageView.setImage(connectivityBitmap.get());
+        } catch (InterruptedException | ExecutionException e) {
+          new Alert(Alert.AlertType.ERROR, "Error creating preview ImageView from provided connectivitySymbol" + e.getMessage()).show();
+        }
+      });
+      
+      
+      // Create a renderer for the associations.
+      UniqueValueRenderer.UniqueValue attachmentUV = new UniqueValueRenderer.UniqueValue("Attachment", "", attachmentSymbol, Arrays.asList(UtilityAssociationType.ATTACHMENT.toString()));
+      UniqueValueRenderer.UniqueValue connectivityUV = new UniqueValueRenderer.UniqueValue("Connectivity", "", connectivitySymbol, Arrays.asList(UtilityAssociationType.CONNECTIVITY.toString()));
+      associationsOverlay.setRenderer(new UniqueValueRenderer(Arrays.asList("AssociationType"), Arrays.asList(attachmentUV, connectivityUV), "", null));
```

Comment:
Let's reorganise the sample a little.
We generally keep the UI setup (symbols, graphics overlays, renderers) before we create/load the map, feature layers and utility networks. That way it's all ready and loaded (hopefully!) before we try and show stuff on the screen.

That way we're also not jumping back and forth between different things.

After that, lets take 90-114 (the block where we load the Utility Network), the interesting part of this sample, and put it after we set up the map view:
```
	>> create the image views/symbols/graphics overlays
	>> create the renderers
	>> create the graphics overlay
    >> add the renderers

  	// create a map with the Topographic Vector Basemap
    ArcGISMap map = new ArcGISMap(Basemap.createTopographicVector());

    // create a viewpoint from envelope
	Viewpoint viewPoint = new Viewpoint(41.8057655, -88.1489692, 23);

    // set initial ArcGISMap extent
    map.setInitialViewpoint(viewPoint);
    
    // create a map view and set its map
    mapView = new MapView();
    mapView.setMap(map);
    mapView.getGraphicsOverlays().add(associationsOverlay);

	>> load the UtilityNetwork and display the associations (lines 90-114)

	// create a controls box

```